doc/operations/code_upgrade.rst

```diff
@@ -0,0 +1,189 @@
+
+.. mermaid::
+
+    graph TB;
+        classDef Primary stroke-width:4px
+
+        subgraph Service
+            Node0((Node 0))
+            Node1((Node 1))
+            class Node1 Primary
+            Node2((Node 2))
+        end
+
+1. First, operators/members should register the new code version corresponding to the new enclave measurement using the ``add_node_code`` proposal action (see :ref:`governance/common_member_operations:Updating Code Version`).
+
+2. The set of new nodes running the enclave registered in the previous step should be added to the service (see :ref:`operations/start_network:Adding a New Node to the Network`) and trusted by members (see :ref:`governance/common_member_operations:Trusting a New Node`). Typically, the same number of nodes than were originally present should be added to the service. In this example, the service is now made of 6 nodes (``f = 2``).
+
+.. mermaid::
+
+    graph TB;
+        classDef NewNode fill:turquoise
+        classDef Primary stroke-width:4px
+
+        subgraph Service
+            subgraph Old Nodes
+                Node0((Node 0))
+                Node1((Node 1))
+                class Node1 Primary
+                Node2((Node 2))
+            end
+
+            subgraph New Nodes
+                Node3((Node 3))
+                Node4((Node 4))
+                Node5((Node 5))
+                class Node3 NewNode
+                class Node4 NewNode
+                class Node5 NewNode
+            end
+        end
+
+
+3. The old code version (i.e. the code version run by nodes 0, 1 and 2) can be removed using the ``remove_node_code`` proposal action.
+
+.. note:: Once the ``remove_node_code`` proposal has successfully been applied, nodes 0, 1 and 2 which run this version of the code will still be running and actively part of the service.
+
+4. The original nodes (``Node 0``, ``Node 1`` and ``Node 2``) can then safely be retired.
+
+- ``Node 0`` is retired, 5 nodes remaining, ``f = 2``:
+
+.. mermaid::
+
+    graph TB;
+        classDef NewNode fill:Turquoise
+        classDef RetiredNode fill:LightGray
+        classDef Primary stroke-width:4px
+
+        Node0((Node 0))
+        class Node0 RetiredNode
+
+        subgraph Service
+            subgraph Old Nodes
+                Node1((Node 1))
+                class Node1 Primary
+                Node2((Node 2))
+            end
+
+            subgraph New Nodes
+                Node3((Node 3))
+                Node4((Node 4))
+                Node5((Node 5))
+                class Node3 NewNode
+                class Node4 NewNode
+                class Node5 NewNode
+            end
+        end
+
+- ``Node 1`` (primary) is retired, 4 nodes remaining, ``f = 1``. ``Node 4`` becomes primary after election phase (during which service cannot temporarily process requests that mutate the state of the key-value store):
```

Comment:
We should make it clear that's true for any election, not specific to a code update.